lambdas/data-migration2/src/granulesAndFiles.ts

```diff
@@ -266,14 +284,25 @@ export const migrateGranulesAndFiles = async (
 
   /* eslint-disable no-await-in-loop */
   while (record) {
-    // eslint-disable-next-line max-len
-    const result = await migrateGranuleAndFilesViaTransaction(record, migrationResult, knex, loggingInterval);
+    const result = await migrateGranuleAndFilesViaTransaction({
+      dynamoRecord: record,
+      granuleAndFileMigrationResult: migrationResult,
+      knex,
+      loggingInterval,
+      stream: errorFileWriteStream,
+    });
     migrationResult.granulesResult = result.granulesResult;
     migrationResult.filesResult = result.filesResult;
 
     await searchQueue.shift();
     record = await searchQueue.peek();
+
+    if (record) {
+      errorFileWriteStream.write(',\n');
+    }
   }
+  errorFileWriteStream.write('\n]}');
```

Comment:
```suggestion
  errorFileWriteStream.end('\n]}');
```

Based on https://nodejs.org/api/stream.html#stream_writable_end_chunk_encoding_callback